File: LeetCode/2125/2125.cpp
class Solution {
public:
    int numberOfBeams(vector<string>& bank) {
        int prev = count(bank[0].begin(), bank[0].end(), '1');
        int answer = 0;
        for(int i{1}; i< bank.size(); ++i)
        {
            int cur = count(bank[i].begin(), bank[i].end(), '1');
            if(cur != 0)
            {
                answer += cur * prev;
                prev = cur;
            }
        }
        return answer;
    }
};

File: LeetCode/2125/2125.rs
impl Solution {
    pub fn number_of_beams(bank: Vec<String>) -> i32 {
        let mut result = 0;
        let mut prev = bank[0].chars().filter(|&x| x == '1').count();
        for i in (1..bank.len())
        {
            let cur = bank[i].chars().filter(|&x| x == '1').count();
            if(cur != 0)
            {
                result += cur * prev;
                prev = cur;
            }
        }
        result as i32

    }
}
